src/sqlfluff/rules/L026.py

```diff
@@ -100,4 +66,84 @@ def _eval(self, context: RuleContext) -> EvalResultType:
         ):
             return LintResult()
 
-        return super()._eval(context=context)
+        violations: List[LintResult] = []
+        start_types = ["select_statement", "delete_statement", "update_statement"]
+        if context.segment.is_type(
+            *start_types
+        ) and not context.functional.parent_stack.any(sp.is_type(*start_types)):
+            dml_target_table: Optional[str] = None
+            if not context.segment.is_type("select_statement"):
+                # Extract first table reference.
+                table_reference = next(
+                    context.segment.recursive_crawl("table_reference"), None
+                )
+                if table_reference:
+                    dml_target_table = table_reference.raw
```

Comment:
I think using raw here means the rule would get confused if the table references were more (or less) qualified in different places?

Bit of a toy example but this raises a lint:
```sql
use schema public;

delete from public.agent1
where exists(
    select customer.cust_id from customer
    where agent1.agent_code <> customer.agent_code);
```

It's a bit of a weird one as it seems niche and relies on the context that we're in the public schema already.
We can't just take the last identifier from the reference as we don't want to not flag:
```sql
delete from schema1.agent1
where exists(
    select customer.cust_id from customer
    where schema2.agent1.agent_code <> customer.agent_code);
```

Reply:
More relevant would be that something like this raises a lint:
```sql
delete from public.agent1
where exists(
    select customer.cust_id from customer
    where public.agent1.agent_code <> customer.agent_code);
```

Reply:
You're right -- I'll take a look at this. Presumably, some existing rule(s) already handle similar cases.